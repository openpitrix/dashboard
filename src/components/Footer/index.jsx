import React, { PureComponent } from 'react';
import { translate } from 'react-i18next';

import styles from './index.scss';

@translate()
export default class Footer extends PureComponent {
  getCurrentLocale = () => typeof window !== 'undefined' && localStorage.getItem('i18nextLng');

  changeLocale = (lang, e) => {
    e.preventDefault();
    if (this.getCurrentLocale() !== lang) {
      this.props.i18n.changeLanguage(lang);
    }
  };

  render() {
    const { t } = this.props;

    return (
      <div className={styles.footer}>
        <div className={styles.wrapper}>
          <div className={styles.inner}>
            <span className={styles.logo}>
              <img src="/logo_grey.svg" alt="logo" height="100%" />
            </span>
            <ul className={styles.terms}>
              <li>
                <a
                  target="_blank"
                  href="https://docs.openpitrix.io/home/"
                  rel="noopener noreferrer"
                >
                  {t('Document')}
                </a>
              </li>
              <li>
                <a
                  target="_blank"
                  href="https://openpitrix.io"
                  rel="noopener noreferrer"
                >
                  {t('About')}
                </a>
                {/* <span className={styles.dot} />
                 <a href="javascript:void(0)">{t('Help')}</a>
                 <span className={styles.dot} />
                 <a href="javascript:void(0)">{t('Terms')}</a> */}
              </li>
              <li>OpenPitrix &copy; 2018</li>
              <li>
                <a
                  href="#"
                  onClick={this.changeLocale.bind(null, 'zh')}
                  className={styles.toggleLocale}
                >
                  中文
                </a>
                <span className={styles.dot} />
                <a
                  href="#"
                  onClick={this.changeLocale.bind(null, 'en')}
                  className={styles.toggleLocale}
                >
                  English
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    );
  }
}
