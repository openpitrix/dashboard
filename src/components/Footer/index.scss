@import "~scss/vars";

$line-height: $footer-height - 1;
$fg-color: #78839e;

.footer {
  position: static;
  width: 100%;
  height: $footer-height;
  background-color: $N0;
  box-sizing: border-box;

  .wrapper {
    width: $content-width;
    margin: 0 auto;
    color: $fg-color;

    .logo {
      display: inline-block;
      float: left;
      font-size: $size-medium;
      line-height: $line-height;
      text-transform: uppercase;
      :global {
        img {
          height: 16px;
        }
      }
    }
    .terms {
      display:inline-block;
      float: right;
      line-height: $line-height;
      & > li{
        list-style-type: none;
        float: left;
        text-align: right;
        margin-left: 10px;

        &.copyright {
          margin-left: 20px;
        }

        a {
          color: $fg-color;
          &:hover {
            color: $P75;
          }
        }
      }
    }
  }

  .dot {
    display:inline-block;
    height: 3px;
    width: 3px;
    background-color: $fg-color;
    border-radius: 50%;
    vertical-align: middle;
  }

  .toggleLocale {
    &:hover {
      color: $P75 !important;
    }
  }
}
