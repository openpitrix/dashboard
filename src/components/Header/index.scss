@import "~scss/vars";

$profile-hover-color: #efe6f8;

@keyframes deepHeader {
  from {
    opacity: 0;
    background: transparent;
  }
  to {
    opacity: 1;
    background: linear-gradient(to left, $P75, #854fb9 32%, #484999);
  }
}

.header {
  position: absolute;
  top: 0;
  width: 100%;
  height: $header-height;
  z-index: 999;
  background: none;

  &.deep {
    background: linear-gradient(to left, $P75, #854fb9 32%, #484999);
  }

  &.deep, &.deepHome {
    position: fixed;

    .login{
      color: $N0;
    }

    .user [class*='target']{
      color: $N0;

      :global{
        .icon-caret-down svg{
          --primary-color: #{$N0};
          --secondary-color: #{$N0};
        }
      }
    }
  }


  &.deepHome {
    animation-duration: 0.5s;
    animation-name: deepHeader;
    animation-iteration-count: 1;
    animation-timing-function: ease-in-out;
    animation-fill-mode: forwards;
  }

  .wrapper {
    width: $content-width;
    height: $header-height;
    margin: 0 auto;
  }

  .logo {
    float: left;
    width: auto;
    height: $logo-height;
    display: inline-block;
    margin-top: $header-height / 2;
    transform: translateY(-50%);
    vertical-align: middle;
  }

  .search {
    float: right;
    display: inline-block;
    min-width: 240px;
    margin-right: 20px;
    margin-top: $header-height / 2;
    transform: translateY(-50%);
    vertical-align: top;

    input {
      box-sizing: border-box;
      width: 240px;
      color: $P45;
      font-weight: 400;
      font-size: $size-normal;
      border-radius: 16px;
      border: 1px solid transparent;
      background-color: rgba($N0, 0.1);
      //transition: color .5s ease-in;
      transition: all .3s ease-in-out;

      &::placeholder {
        font-size: 14px;
        color: $P45;
        line-height: 1.0;
        letter-spacing: 0;
      }

      &:focus {
        transition: width .2s ease-in;
        width: 360px;
        font-weight: normal;
        color: $N0;
        border: solid 1px rgba($N0, 0.5) !important;
      }
      &:hover {
        color: $N0;
        border: solid 1px rgba($N0, 0.3);
      }
    }

    :global {
      .qicon {
        --primary-color: #fff;
        //--secondary-color: #d3c4ed;
      }
      .icon {
        color: $N0;
        opacity: 0.5;
        &:hover, &:focus {
          opacity: 1;
          svg {
            --primary-color: #fff;
          }
        }
      }
    }
  }

  .login{
    float: right;
    display: inline-block;
    margin: 14px 0;
    line-height: 20px;
    color: $N500;
    font-weight: 500;
    opacity: 0.9;

    &:hover{
      opacity: 1;
    }
  }

  .user [class*='target']{
    float: right;
    margin: 12px 24px;
    height: 20px;
    width: auto;
    line-height: 20px;
    opacity: 0.9;
    color: $N500;
    font-weight: 500;
    border: 0 none;

    &:hover{
      border: 0 none;
      box-shadow: none;
      opacity: 1;
    }

    :global{
      .icon-caret-down{
        position: relative;
        top: 3px;
        left: 2px;
        svg {
          --primary-color: #{$N500};
          --secondary-color: #{$N500};
        }
      }
    }
  }

  .operateItems{
    min-width: 120px;
    li a{
      display: block;
      height: 32px;
      padding: 0 12px;
      //width: 100%;
      line-height: 32px;
      font-size: $font14;
      font-weight: normal;
      color: $N300;
      border-radius: 0;
      &:hover {
        background-color: $N10;
        color: $N500;
      }
      &.line{
        border-bottom: 1px solid $N10;
      }
    }
  }

  :global{
    .pi-popover-popper{
      margin-left: -62px !important;
      margin-top: 6px !important;
    }
  }
}

.menuOuter {
  .menus{
    float: left;
    margin-left: 50px;
    a{
      display: inline-block;
      margin: 0 12px;
      padding: 13px 0;
      font-size: 14px;
      font-weight: 500;
      line-height: 20px;
      color: $N0;
      &.active{
        border-bottom: 2px solid $Y200;
      }
    }
  }
}
