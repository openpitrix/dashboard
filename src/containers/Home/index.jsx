import React, { Component } from 'react';
import { observer, inject } from 'mobx-react';

import Nav from 'components/Nav';
import AppList from 'components/AppList';
import styles from './index.scss';

@inject('rootStore')
@observer
export default class Home extends Component {
  static async onEnter({ appStore }) {
    await appStore.fetchApps();
  }

  render() {
    const { config, appStore } = this.props.rootStore;

    return (
      <div className={styles.home}>
        <Nav className={styles.nav} navs={config.navs}/>
        <AppList className={styles.apps} apps={appStore.apps}/>
      </div>
    );
  }
}
