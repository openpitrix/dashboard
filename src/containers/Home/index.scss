@import "~scss/variables";

.home{
  display: table;
  width: 100%;
  height: calc(100vh - #{$footer-height});
  .nav {
    display: table-cell;
    width: 210px;
    padding-left: calc(50vw - #{$content-width}/2);
    border-right: 1px solid $border;
    background: #ffffff;
    box-sizing: content-box;
    vertical-align: top;
  }
  .apps {
    display: table-cell;
    vertical-align: top;
    padding-right: calc(50vw - #{$content-width}/2);
  }
}