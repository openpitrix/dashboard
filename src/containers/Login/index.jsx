import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { observer, inject } from 'mobx-react';
import { throttle } from 'lodash';

import Logo from 'components/Logo';
import Form from 'components/Base/Form';
import Input from 'components/Base/Input';
import Button from 'components/Base/Button';
import Checkbox from 'components/Base/Checkbox';
import Notification from 'components/Base/Notification';

import styles from './index.scss';

@inject(({ rootStore }) => ({
  store: rootStore.loginStore
}))
@observer
export default class Login extends Component {
  handleSubmit = params => {
    this.props.store.login(params);
  };

  resetErrMsg = () => {
    this.props.store.errMsg = '';
  };

  renderErrMsg() {
    let { errMsg } = this.props.store;

    if (errMsg) {
      return <Notification message={errMsg} onHide={this.resetErrMsg} />;
    }
    return null;
  }

  render() {
    return (
      <div className={styles.login}>
        <div className={styles.loginTitle}>
          <Logo url="/assets/logo3.svg" />
        </div>
        <div className={styles.loginForm}>
          <h1>Login OpenPitrix</h1>
          <Form onSubmit={throttle(this.handleSubmit, 1000)}>
            <Form.Item className={styles.formItem}>
              <Input
                className={styles.formInput}
                icon="search"
                name="username"
                placeholder="Username"
              />
            </Form.Item>
            <Form.Item className={styles.formItem}>
              <Input
                className={styles.formInput}
                type="password"
                icon="search"
                name="password"
                placeholder="Password"
              />
            </Form.Item>
            <Form.Item className={styles.formItem}>
              <Checkbox className={styles.checkbox}>Remember me</Checkbox>
            </Form.Item>
            <Form.Item className={styles.formItem}>
              <Button htmlType="submit" className={styles.submitBtn}>
                Login
              </Button>
            </Form.Item>
          </Form>
        </div>
        <p className={styles.tip}>
          <span>No openpiotrix account?</span>
          <Link to="/signup">Sign up</Link>
        </p>

        {this.renderErrMsg()}
      </div>
    );
  }
}
