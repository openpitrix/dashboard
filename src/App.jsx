import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import { Switch, Route, Redirect } from 'react-router-dom';
import { Provider } from 'mobx-react';

import Header from 'components/Header';
import Footer from 'components/Footer';
import RouteWrapper from './routes/wrapper';
import routes from './routes';
import { getCookie } from './utils';

import './scss/main.scss';

class App extends Component {
  static propTypes = {
    rootStore: PropTypes.object
  };

  hasLoggedIn = () => {
    return !!getCookie('user');
  };

  render() {
    const { location, rootStore } = this.props;
    const pathname = location.pathname;

    const isHome = pathname === '/' || pathname === '/apps';
    const isLogin = pathname === '/login';

    return (
      <Provider rootStore={rootStore}>
        <Fragment>
          {!isLogin && <Header isHome={isHome} />}
          <div className="main">
            <Switch>
              {routes.map((route, i) => (
                <Route
                  key={i}
                  exact={route.exact}
                  path={route.path}
                  render={({ match }) =>
                    route.needAuth && !this.hasLoggedIn() ? (
                      <Redirect to="/login" />
                    ) : (
                      <RouteWrapper
                        component={route.component}
                        match={match}
                        rootStore={rootStore}
                      />
                    )
                  }
                />
              ))}
            </Switch>
          </div>
          {!isLogin && <Footer />}
        </Fragment>
      </Provider>
    );
  }
}

export default withRouter(App);
