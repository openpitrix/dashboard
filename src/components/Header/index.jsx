import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { NavLink } from 'react-router-dom';
import { observer, inject } from 'mobx-react';
import { ButtonDropdown, DropdownToggle, DropdownItem, DropdownMenu } from 'reactstrap';
import { getLoginUser } from 'src/utils';

import Logo from '../Logo';
import Input from '../Base/Input';
import styles from './index.scss';

@inject('rootStore')
@observer
export default class Header extends Component {
  state = {
    isHome: PropTypes.bool,
    dropdownOpen: false
  };

  toggleDropdown = () => {
    this.setState({
      dropdownOpen: !this.state.dropdownOpen
    });
  };

  renderLoginButton() {
    const loggeInUser = getLoginUser();

    if (!loggeInUser) {
      return <NavLink to="/login">Sign In</NavLink>;
    }

    return (
      <ButtonDropdown
        className={styles.loginButton}
        isOpen={this.state.dropdownOpen}
        toggle={this.toggleDropdown}
      >
        <DropdownToggle caret className={styles.toggleBtn}>
          {loggeInUser}
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem style={{ padding: 0 }}>
            <a href="/logout">Logout</a>
          </DropdownItem>
        </DropdownMenu>
      </ButtonDropdown>
    );
  }

  render() {
    const {
      isHome,
      rootStore: { fixNav }
    } = this.props;
    const isDark = !isHome || fixNav;
    const logoUrl = isDark ? '/assets/logo2.svg' : '/assets/logo.svg';

    return (
      <div className={classnames('header', styles.header, { [styles.darkHeader]: !isDark })}>
        <div className={styles.wrapper}>
          <Logo className={styles.logo} url={logoUrl} />
          {isDark && (
            <Input.Search className={styles.search} placeholder="Search apps in Pitrix..." />
          )}
          <div className={styles.menus}>
            <NavLink to="/apps">Catalog</NavLink>
            <NavLink to="/deployment">Deployment</NavLink>
            <NavLink to="/manage">{getLoginUser() ? 'Manage' : 'Development'}</NavLink>
            {this.renderLoginButton()}
          </div>
        </div>
      </div>
    );
  }
}
