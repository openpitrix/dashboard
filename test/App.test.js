import { observable } from 'mobx';
import { Provider } from 'mobx-react';
// import { withRouter } from 'react-router';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import App from 'src/App';
// import routes from 'src/routes';
import Home from 'pages/Home';
import Login from 'pages/Login';
import Header from 'components/Header';
import Footer from 'components/Footer';

const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  clear: jest.fn()
};
global.localStorage = localStorageMock;

const renderRoute = () => {};

const renderPage = (routes, location, store) => (
  <Provider rootStore={store}>
    <BrowserRouter>
      <App location={location}>
        <Switch>
          {routes.map((route, i) => (
            <Route
              key={i}
              exact={route.exact}
              path={route.path}
              render={({ match }) => renderRoute(match, route, store)}
            />
          ))}
        </Switch>
      </App>
    </BrowserRouter>
  </Provider>
);

describe('<App/>', () => {
  let store,
    location,
    routes;

  beforeEach(() => {
    store = observable.object({
      fixNav: false,
      appStore: {
        appSearch: () => {}
      }
    });

    // mock routes
    routes = [
      { path: '/', component: Home, exact: true },
      { path: '/login', component: Login, exact: true }
    ];
  });

  afterEach(() => {
    store = null;
    routes = [];
  });

  xit('basic render', () => {
    location = { pathname: '/' };
    const wrapper = render(renderPage(routes, location, store));
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  xit('body has main className', () => {
    const wrapper = shallow(<App rootStore={store} />);
    expect(wrapper.find('.main').length).toBe(1);
  });

  xit('/login page does not have Header and Footer', () => {
    location = { pathname: '/login' };
    const wrapper = shallow(<App rootStore={store} location={location} />);
    expect(wrapper.find(Header).length).toBe(0);
    expect(wrapper.find(Footer).length).toBe(0);
  });
});
