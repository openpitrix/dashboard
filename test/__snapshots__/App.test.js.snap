// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> basic render 1`] = `
<div
  class="main"
/>
`;
